Feature/tests

dzil123 opened the pull request.

dzil123 committed:
W: Changed behavior NodeTreeImportError real_node
Continuation of bugfix/cleanup.

dzil123 committed:
T: Start adding importnodetree() tests

dzil123 committed:
B: Create setup.cfg for shared config of tools
Add pytest to requirements, add pytest config

dzil123 committed:
T: Make mock run automatically, finish first test

132ikl commented:
re-request review when you want

dzil123 committed:
T: Add pytest github action

dzil123 commented:
TODO:
- ~~dont install suggested or nonoptional dependencies~~
- ~~figure out how to parse requirements_extra.txt~~

All done!

132ikl commented on .github/workflows/format.yml:
```diff
@@ -19,6 +19,7 @@ jobs:
       - name: Install formatters
         run: |
           python -m pip install --upgrade pip
+          python -m pip install --upgrade setuptools wheel
           pip install autoflake black isort
```
You can combine these, or at least choose between `python3 -m pip` vs `pip`

dzil123 replied:
[Github](https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#requirements-file) recommends splitting up these steps:
```bash
python -m pip install --upgrade pip
pip install -r requirements.txt
```
I think `python3` is redundant compared to `python` because `actions/setup-python@v1` takes care of putting the correct version in the PATH (the containers come with many versions preinstalled).

132ikl replied:
splitting up pip upgrade and install is fine but why are setuptools and wheel separate from autoflake, black, isort

dzil123 replied:
> why are setuptools and wheel separate from autoflake, black, isort

Because the updating of pip, setuptools, and wheel is a shared set of common tasks to do when setting up a new python environment, and I want to seperate the shared tasks from the unique tasks.

Maybe it would make sense to split up this step into two steps.

dzil123 committed:
B: Revamp requirements to json format

132ikl committed:
B: Add overlay install script

132ikl committed:
B: Refactor requirements.py, include requirements.txt by default

dzil123 committed:
B, T: fix and add features to requirements.py

dzil123 commented:
If there are no more changes to be made to the structure of tests or the new requirements system, I think this is ready to merge into dev. We can always add more tests later.

132ikl commented on requirements.txt:
```diff
@@ -1,20 +1,19 @@
-Jinja2==2.11.1
+httptools==0.1.1
+Jinja2==2.11.2
 MarkupSafe==1.1.1
```
Downgrade this as discussed

dzil123 committed:
B: Update all requirements
Did not update fastapi and dependencies: testing needed
Did not update httptools, numpy to not slow down packages build
Downgrade MarkupSafe to match current behavior of packages build

dzil123 commented:
I force pushed to squash some commits, make the requested requirements modifications, and modify commit description in order to avoid a squash merge to dev.

132ikl commented on requirements.py:
```diff
@@ -0,0 +1,197 @@
+    return defaults
+
+
+parser.add_argument(
+    "overlays",
+    action="append",
+    nargs="*",
+    help="overlays to install",
+    default=get_default_overlays(),
+)
```
This should be moved above `get_default_overlays`.

dzil123 replied:
That won't work because the function is called here. `parser.add_argument` default parameter is not a function that returns a value, I just call a function to have a local scope where I create the default value.
